### PR TITLE
Phase 6: 4-Layer Surface Refine on Asinh Baseline (srf4L)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
PR #2074 showed srf4L (surface_refine_layers=4) as the strongest individual model in the diverse ensemble. Test with multi-seed on the asinh baseline.

| GPU 0-3 | Baseline (3-layer refine) seeds 42-45 |
| GPU 4-7 | srf4L (4-layer refine) seeds 42-45 |

`--surface_refine_layers 4` + all asinh baseline flags.

Baseline: p_in 13.03, p_oodc 7.83, p_tan 30.29, p_re 6.45

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2079 because ADVISOR: Closing — srf4L (4-layer surface refine) is clearly WORSE than the 3-layer baseline.

**W&B analysis (phase6/srf4L-multiseed, 8 finished runs):**

| Metric | 3L Mean (n=4) | srf4L Mean (n=4) | Change |
|--------|--------------|------------------|--------|
| p_in | 13.30 | 13.29 | -0.1% (flat) |
| p_oodc | 7.78 | 7.83 | +0.7% |
| **p_tan** | **30.33** | **32.01** | **+5.5% WORSE** |
| p_re | 6.46 | 6.53 | +1.1% |

The 4th surface refine layer hurts tandem transfer generalization significantly. Fern's independent 8-seed validation (#2083) confirms: p_tan +7.0% worse with srf4L.

This invalidates the srf4L ensemble strategy. Reassigning you to a new experiment shortly..

**Comment by tcapelle** [comment]
STUDENT askeladd:

## Results

### Baseline (surface_refine_layers=3) — Seeds 42-45

| Seed | val/loss | p_in | p_oodc | p_tan | p_re | W&B Run |
|------|----------|------|--------|-------|------|---------|
| 42 | 0.3897 | 13.6 | 7.7 | 30.7 | 6.6 | hfyp045n |
| 43 | 0.3901 | 13.0 | 7.8 | 30.5 | 6.5 | cwnsbzq3 |
| 44 | 0.3905 | 13.7 | 7.7 | 30.1 | 6.5 | j6nx4oje |
| 45 | 0.3874 | 12.9 | 7.8 | 30.1 | 6.3 | ll7rta5t |
| **Mean** | **0.3894** | **13.30** | **7.75** | **30.35** | **6.48** | |
| **Std** | **0.0014** | **0.39** | **0.06** | **0.29** | **0.13** | |

### srf4L (surface_refine_layers=4) — Seeds 42-45

| Seed | val/loss | p_in | p_oodc | p_tan | p_re | W&B Run |
|------|----------|------|--------|-------|------|---------|
| 42 | 0.3908 | 12.9 | 7.8 | 31.2 | 6.3 | dvktxv1k |
| 43 | 0.3950 | 13.7 | 8.0 | 31.7 | 6.7 | bfh72bq0 |
| 44 | 0.3969 | 13.4 | 7.8 | 32.1 | 6.6 | wjwqpkj8 |
| 45 | 0.4013 | 13.3 | 7.7 | 33.0 | 6.5 | 9b7cdfdb |
| **Mean** | **0.3960** | **13.33** | **7.83** | **32.00** | **6.53** | |
| **Std** | **0.0044** | **0.33** | **0.13** | **0.76** | **0.17** | |

### Comparison (srf4L vs Baseline)

| Metric | Baseline (3L) | srf4L (4L) | Delta | % Change |
|--------|--------------|------------|-------|----------|
| val/loss | 0.3894 | 0.3960 | +0.0066 | +1.7% |
| p_in | 13.30 | 13.33 | +0.03 | +0.2% |
| p_oodc | 7.75 | 7.83 | +0.08 | +1.0% |
| p_tan | 30.35 | 32.00 | +1.65 | **+5.4%** |
| p_re | 6.48 | 6.53 | +0.05 | +0.8% |

### PR Baseline Comparison

PR baseline: p_in 13.03, p_oodc 7.83, p_tan 30.29, p_re 6.45

- Baseline 3L (this run): Very close to PR baseline — within noise (p_in 13.30 vs 13.03, p_tan 30.35 vs 30.29)
- srf4L 4L: Worse across the board, especially p_tan (+5.4%)

### Commands Used

```bash
# Baseline (GPUs 0-3)
CUDA_VISIBLE_DEVICES={0,1,2,3} python train.py --agent askeladd --wandb_name "askeladd/srf4L-baseline-s{42-45}" \
  --wandb_group "phase6/srf4L-multiseed" --asinh_pressure --asinh_scale 0.75 \
  --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp \
  --n_layers 3 --slice_num 96 --tandem_ramp \
  --domain_layernorm --domain_velhead --ema_decay 0.999 \
  --weight_decay 5e-5 --cosine_T_max 160 --disable_pcgrad \
  --pressure_first --pressure_deep --residual_prediction \
  --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 --seed {42-45}

# srf4L (GPUs 4-7)
CUDA_VISIBLE_DEVICES={4,5,6,7} python train.py --agent askeladd --wandb_name "askeladd/srf4L-4layer-s{42-45}" \
  --wandb_group "phase6/srf4L-multiseed" --asinh_pressure --asinh_scale 0.75 \
  --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp \
  --n_layers 3 --slice_num 96 --tandem_ramp \
  --domain_layernorm --domain_velhead --ema_decay 0.999 \
  --weight_decay 5e-5 --cosine_T_max 160 --disable_pcgrad \
  --pressure_first --pressure_deep --residual_prediction \
  --surface_refine --surface_refine_hidden 192 --surface_refine_layers 4 --seed {42-45}
```

W&B group: `phase6/srf4L-multiseed`

### What Happened

**The 4-layer surface refinement is a clear negative.** Adding a 4th hidden layer to the surface refinement MLP hurts performance across all metrics, most significantly on tandem transfer (p_tan +5.4%). The val/loss is also worse (+1.7%) with higher variance (std 0.0044 vs 0.0014).

The srf4L variant was identified as the "strongest individual model" in the diverse ensemble (PR #2074), but this was likely due to it contributing diversity rather than being intrinsically better. When tested head-to-head with controlled seeds on the asinh baseline, the extra layer clearly overfits — particularly visible on the tandem transfer split where it generalizes poorly.

The baseline 3-layer configuration remains optimal for the surface refinement MLP.

### Suggested Follow-ups

- The extra capacity from srf4L may still have value in ensembles (contributing diversity), but it should not replace the 3-layer default
- If more surface refinement capacity is desired, increasing `surface_refine_hidden` (e.g. 256) while keeping 3 layers might be a better path than adding depth
- Could also try surface_refine_layers=4 with dropout or weight decay specific to the refinement head to control overfitting
